plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/shipping/shipping-address.tsx

```diff
@@ -4,82 +4,87 @@
 import PickupLocation from '@woocommerce/base-components/cart-checkout/pickup-location';
 import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
 import { useSelect } from '@wordpress/data';
-
-/**
- * Internal dependencies
- */
-import ShippingLocation from '../../shipping-location';
-import { CalculatorButton, CalculatorButtonProps } from './calculator-button';
+import { useEffect, useState } from '@wordpress/element';
 
 export interface ShippingAddressProps {
-	showCalculator: boolean;
-	isShippingCalculatorOpen: boolean;
-	setIsShippingCalculatorOpen: CalculatorButtonProps[ 'setIsShippingCalculatorOpen' ];
+	setShippingCalculatorLabel: React.Dispatch<
+		React.SetStateAction< string >
+	>;
+	setShippingCalculatorAddress: React.Dispatch<
+		React.SetStateAction< string >
+	>;
 	shippingAddress: ShippingAddressType;
+	hasRates: boolean;
 }
 
-export type ActiveShippingZones = {
-	description: string;
-}[];
-
 export const ShippingAddress = ( {
-	showCalculator,
-	isShippingCalculatorOpen,
-	setIsShippingCalculatorOpen,
+	setShippingCalculatorLabel,
+	setShippingCalculatorAddress,
 	shippingAddress,
+	hasRates,
 }: ShippingAddressProps ): JSX.Element | null => {
 	const { isEditor } = useEditorContext();
 	const prefersCollection = useSelect( ( select ) =>
 		select( CHECKOUT_STORE_KEY ).prefersCollection()
 	);
-	const activeShippingZones: ActiveShippingZones = getSetting(
-		'activeShippingZones'
-	);
 
-	const hasMultipleAndDefaultZone =
-		activeShippingZones.length > 1 &&
-		activeShippingZones.some(
-			( zone: { description: string } ) =>
-				zone.description === 'Everywhere' ||
-				zone.description === 'Locations outside all other zones'
-		);
+	const [ showPickupLocation, setShowPickupLocation ] = useState( false );
+
+	useEffect( () => {
+		const hasFormattedAddress = !! formatShippingAddress( shippingAddress );
 
-	const hasFormattedAddress = !! formatShippingAddress( shippingAddress );
+		// If it's the editor and store has no formatted address,
+		// then the shipping calculator labels will not be modified.
+		// Shipping calculator will display the default label.
+		if ( isEditor && ! hasFormattedAddress ) {
+			return;
+		}
 
-	// If there is no default customer location set in the store,
-	// and the customer hasn't provided their address,
-	// and only one default shipping method is available for all locations,
-	// then the shipping calculator will be hidden to avoid confusion.
-	if ( ! hasFormattedAddress && ! isEditor && ! hasMultipleAndDefaultZone ) {
-		return null;
+		if ( prefersCollection ) {
+			setShowPickupLocation( true );
+			return;
+		}
+
+		setShowPickupLocation( false );
+
+		const formattedAddress = formatShippingAddress( shippingAddress );
+		let deliveryLabel;
+		if ( hasRates ) {
+			deliveryLabel = __( 'Delivers to', 'woocommerce' );
+		} else {
+			deliveryLabel = __(
+				'No delivery options available for',
+				'woocommerce'
+			);
+		}
```

Comment:
Nitpic: We could condense this part into the following three-liner:
```suggestion
		const deliveryLabel = hasRates
			? __( 'Delivers to', 'woocommerce' )
			: __( 'No delivery options available for', 'woocommerce' );
```